src/main/kotlin/no/nav/familie/oppdrag/konsistensavstemming/KonsistensavstemmingMapper.kt

```diff
@@ -0,0 +1,148 @@
+            henvisning = utbetalingsperiode.behandlingId.toString()
+            attestantListe.add(lagAttestant(utbetalingsoppdrag))
+        }
+    }
+
+    private fun akkumulerTotalbeløp(utbetalingsperiode: Utbetalingsperiode) {
+        // utlede om utbetalingsperioden er aktuell for avstemmingsdato
+        if (utbetalingsperiode.vedtakdatoFom.isBefore(avstemmingsDato.toLocalDate()) && utbetalingsperiode.vedtakdatoTom.isAfter(avstemmingsDato.toLocalDate())) {
+            totalBeløp+=utbetalingsperiode.sats.toLong()
+        }
+    }
+
+    private fun lagAttestant(utbetalingsoppdrag: Utbetalingsoppdrag): Attestant {
+        return Attestant().apply {
+            attestantId = utbetalingsoppdrag.saksbehandlerId
+        }
+    }
+
+    private fun lagEnhet(): Enhet {
+        return Enhet().apply {
+            enhetType = OppdragSkjemaConstants.ENHET_TYPE
+            enhet = OppdragSkjemaConstants.ENHET
+            enhetFom = OppdragSkjemaConstants.ENHET_DATO_FOM.format(datoFormatter)
+        }
+    }
+
+    private fun lagTotaldata(totalantall: Int): Konsistensavstemmingsdata {
+        val konsistensavstemmingsdata = lagAksjonsmelding(KonsistensavstemmingConstants.DATA)
+        konsistensavstemmingsdata.apply {
+            totaldata = Totaldata().apply {
+                totalAntall = totalantall.toBigInteger()
+                totalBelop = BigDecimal.valueOf(totalBeløp)
+                fortegn = getFortegn(totalBeløp)
+            }
+        }
+        return konsistensavstemmingsdata
+    }
+
+    private fun getFortegn(satsbeløp: Long): String {
+        return if (BigDecimal.valueOf(satsbeløp) >= BigDecimal.ZERO) KonsistensavstemmingConstants.FORTEGN_T else KonsistensavstemmingConstants.FORTEGN_F
+    }
+
+    private fun lagAksjonsmelding(aksjontype: String): Konsistensavstemmingsdata =
+            Konsistensavstemmingsdata().apply {
+                aksjonsdata = opprettAksjonsdata(aksjontype)
+            }
+
+    private fun opprettAksjonsdata(aksjonstype: String): Aksjonsdata {
+        return Aksjonsdata().apply {
+            this.aksjonsType = aksjonstype
+            this.kildeType = KonsistensavstemmingConstants.KILDETYPE
+            this.avstemmingType = KonsistensavstemmingConstants.KONSISTENSAVSTEMMING
+            this.avleverendeKomponentKode = fagsystem
+            this.mottakendeKomponentKode = KonsistensavstemmingConstants.OS
+            this.underkomponentKode = fagsystem
+            this.tidspunktAvstemmingTom = avstemmingsDato.format(tidspunktFormatter)
+            this.avleverendeAvstemmingId = avstemmingId
+            this.brukerId = fagsystem
+        }
+    }
+
+    private fun encodeUUIDBase64(uuid: UUID): String {
+        val bb = ByteBuffer.wrap(ByteArray(16))
+        bb.putLong(uuid.mostSignificantBits)
+        bb.putLong(uuid.leastSignificantBits)
+        return Base64.getUrlEncoder().encodeToString(bb.array()).substring(0, 22)
+    }
```

Comment:
Er denne metoden lik som for grensesnittavstemming? :) Kanskje vi kan lage en abstrakt mapperklasse med noen default-metoder som mapperne våre kan arve, hvis det er flere metoder som denne? Ble litt vanskelig med et interface siden xsd-ene er ulike...